firmware/controllers/trigger/trigger_central.cpp

```diff
@@ -686,28 +686,32 @@ static void resetRunningTriggerCounters() {
 
 void onConfigurationChangeTriggerCallback(DECLARE_ENGINE_PARAMETER_SIGNATURE) {
 	bool changed = false;
-	for (int i = 0; i < CAM_INPUTS_COUNT; i++) {
+	for (int i = 0; i < efi::size(CONFIG(camInputs)); i++) {
 		changed |= isConfigurationChanged(camInputs[i]);
 	}
 
-	for (int i = 0; i < GAP_TRACKING_LENGTH; i++) {
+	for (int i = 0; i < efi::size(CONFIG(triggerGapOverride)); i++) {
 		changed |= isConfigurationChanged(triggerGapOverride[i]);
 	}
 
-	changed |=
-		isConfigurationChanged(trigger.type) ||
-		isConfigurationChanged(ambiguousOperationMode) ||
-		isConfigurationChanged(useOnlyRisingEdgeForTrigger) ||
-		isConfigurationChanged(globalTriggerAngleOffset) ||
-		isConfigurationChanged(trigger.customTotalToothCount) ||
-		isConfigurationChanged(trigger.customSkippedToothCount) ||
-		isConfigurationChanged(triggerInputPins[0]) ||
-		isConfigurationChanged(triggerInputPins[1]) ||
-		isConfigurationChanged(triggerInputPins[2]) ||
-		isConfigurationChanged(vvtMode) ||
```

Comment:
This line, `isConfigurationChanged(vvtMode)`, was the problem.  If you compare arrays with identical contents, they will compare not equal since it will actually compare the _pointer_, not the values within the array.